core/trino-spi/src/test/java/io/trino/spi/TestSpiBackwardCompatibility.java

```diff
@@ -42,34 +42,33 @@
-                    "Method: public io.trino.spi.ptf.TableArgumentSpecification$Builder io.trino.spi.ptf.TableArgumentSpecification$Builder.passThroughColumns(boolean)",
-                    "Class: public abstract class io.trino.spi.ptf.ConnectorTableFunction",
-                    "Constructor: public io.trino.spi.ptf.ConnectorTableFunction(java.lang.String,java.lang.String,java.util.List<io.trino.spi.ptf.ArgumentSpecification>,io.trino.spi.ptf.ReturnTypeSpecification)",
-                    "Method: public java.util.List<io.trino.spi.ptf.ArgumentSpecification> io.trino.spi.ptf.ConnectorTableFunction.getArguments()",
-                    "Method: public io.trino.spi.ptf.ReturnTypeSpecification io.trino.spi.ptf.ConnectorTableFunction.getReturnTypeSpecification()",
-                    "Method: public java.lang.String io.trino.spi.ptf.ConnectorTableFunction.getName()",
-                    "Method: public java.lang.String io.trino.spi.ptf.ConnectorTableFunction.getSchema()"))
-            .put("383", ImmutableSet.of(
-                    "Method: public abstract java.lang.String io.trino.spi.function.AggregationState.value()",
-                    "Method: public default void io.trino.spi.security.SystemAccessControl.checkCanExecuteFunction(io.trino.spi.security.SystemSecurityContext,io.trino.spi.connector.CatalogSchemaRoutineName)",
-                    "Method: public default void io.trino.spi.connector.ConnectorAccessControl.checkCanExecuteFunction(io.trino.spi.connector.ConnectorSecurityContext,io.trino.spi.connector.SchemaRoutineName)"))
-            .put("384", ImmutableSet.of(
-                    "Constructor: public io.trino.spi.eventlistener.QueryInputMetadata(java.lang.String,java.lang.String,java.lang.String,java.util.List<java.lang.String>,java.util.Optional<java.lang.Object>,java.util.OptionalLong,java.util.OptionalLong)"))
-            .buildOrThrow();
+            // example
+            .put("123", "Class: public static class io.trino.spi.predicate.BenchmarkSortedRangeSet$Data")
+            // example
+            .put("123", "Constructor: public io.trino.spi.predicate.BenchmarkSortedRangeSet$Data()")
+            // example
+            .put("123", "Method: public void io.trino.spi.predicate.BenchmarkSortedRangeSet$Data.init()")
+            // example
+            .put("123", "Field: public java.util.List<io.trino.spi.predicate.Range> io.trino.spi.predicate.BenchmarkSortedRangeSet$Data.ranges")
+            .put("377", "Constructor: public io.trino.spi.memory.MemoryPoolInfo(long,long,long,java.util.Map<io.trino.spi.QueryId, java.lang.Long>,java.util.Map<io.trino.spi.QueryId, java.util.List<io.trino.spi.memory.MemoryAllocation>>,java.util.Map<io.trino.spi.QueryId, java.lang.Long>)")
+            .put("382", "Method: public io.trino.spi.ptf.TableArgumentSpecification$Builder io.trino.spi.ptf.TableArgumentSpecification$Builder.rowSemantics(boolean)")
```

Comment:
Should we remove old entries? Or should we remove examples if we would like to keep the history of backward incompatible changes?

Reply:
> Should we remove old entries?

i now the comment intructs so, but this is orthogonal to this PR.